docs/designingProcesses/BasicNodes.md

```diff
@@ -9,94 +17,79 @@ Records from the `source` which meet filter's condition go to the `true sink`, a
-
-
-
-# Split 
-Each output receives all records and processes them independently. 
-
-![split graph](../img/split_graph.png)
-
-Every record from the `source` gos to `sink 1` and `sink 2`.
-
-### Parameters
-Split doesn't have additional parameters.
-
-![split window](../img/split_window.png)
-
-### Flow
-Split has one or more outputs. Outputs have the same record as input, so all outputs are identical.
-
+The Expression field should contain the SpEL expression for the filtering conditiona and should produce a boolean value
 
 
 # Switch
-Distributes incoming records between outputs.
+   
+Switch distributes incoming records among output branches in accordance with the criteria configured in those branches
  
 ![switch graph](../img/switch_graph.png)
 
-Each record form the `source` is tested against condition defined on the edge. If `color` is `blue` record goes to the `blue sink`.  If `color` is `green` record goes to the `green sink`. For every other value record goes to the `sink for others`.
+Each record form the `source` is tested against condition defined on the edge. If `#color` is `blue` record goes to the `blue sink`.  If `#color` is `green` record goes to the `green sink`. For every other value record goes to the `sink for others`.
 
-### Parameters 
 ![switch window](../img/switch_window.png)
 
-There are two parameters `Expression` and `exprVal`. `Expression` contains expression which is evaluated for each record, and evaluated value is assigned to variable named in `exprVal`.
- 
-Edges outgoing from `Switch` node can be of one of two types.
+There are two parameters `Expression` and `exprVal`. `Expression` contains expression which is evaluated for each record, and result is assigned to variable named in `exprVal` - `#color` in the example above.
  
 ![switch_edge_condition](../img/switch_edge_condition.png)
 
-Edge of the `Condition` type has an `Expression`. Record go to the first output with matching condition. Order of matching outgoing edges is not guaranteed.
+Eeach edge outgoing from `Switch` node has a boolean expression attached to it; if the expression evaluates to true the record is allowed to pass through this edge.  Record go to the first output with matching condition. *Order of matching outgoing edges is not guaranteed.*
 
 ![switch_edge_default](../img/switch_edge_default.png)
 
 There can be at most one edge of type `Default`, and it gets all records that don't match any `Condition` edge. 
 
+
 ### Flow
 For each incoming record `Expression` is evaluated and result is assigned to `exprVal` variable. After that, records are tested against condition `Expressions` from output edges one by one. Record goes to first output which condition it satisfies. If record doesn't match any conditional output, and default output exists, record goes to default output. Otherwise mismatching record is filtered out.
 
 
+## SQL Variable
```

Comment:
I think it's a bit involved what this node does... I wonder if it's better to hide/delete it...

Reply:
It is such an important component, I would not.